cli/azd/extensions/azure.ai.agents/internal/pkg/agents/agent_api/logging_policy.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package agent_api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"maps"
+	"net/http"
+	"os"
+	"slices"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+// loggingPolicy is a custom pipeline policy that logs HTTP requests and responses
+type loggingPolicy struct {
+	logFile *os.File
+}
+
+// NewLoggingPolicy creates a new logging policy that writes to the specified file
+func NewLoggingPolicy(logFile *os.File) policy.Policy {
+	return &loggingPolicy{logFile: logFile}
+}
+
+// Do implements the policy.Policy interface
+func (p *loggingPolicy) Do(req *policy.Request) (*http.Response, error) {
+	p.logRequest(req)
+
+	resp, err := req.Next()
+	if err != nil {
+		p.logError(err)
+		return resp, err
+	}
+
+	p.logResponse(resp)
+
+	return resp, nil
+}
+
+func (p *loggingPolicy) logRequest(req *policy.Request) {
```

Comment:
We should ensure we are redacting any sensitive information. Review azd logging policy for examples.